### PR TITLE
Remove namespace from ResourceReference

### PR DESCRIPTION
**Pull request by matthchr**
As per https://github.com/kubernetes/community/pull/5455, it is not a recommended pattern.

### CODE REVIEW

File: v2/pkg/genruntime/resource_reference.go [hunk 1]
```diff
@@ -111,6 +109,13 @@ func (ref ResourceReference) Validate() error {
 	return nil
 }
 
+func (ref ResourceReference) ToNamespacedRef(namespace string) NamespacedResourceReference {
```
Comment (babbageclunk) [inline v2/pkg/genruntime/resource_reference.go:112]: Maybe add a test for this? I guess it's covered by other things though.

File: v2/tools/generator/internal/interfaces/kubernetes_resource_interface.go [hunk 2]
```diff
@@ -400,7 +400,7 @@ func createResourceReference(
 			"ResourceReference"))
 	compositeLitDetails.AddField("Group", group)
 	compositeLitDetails.AddField("Kind", kind)
-	compositeLitDetails.AddField("Namespace", astbuilder.Selector(dst.NewIdent(receiverIdent), "Namespace"))
+	// compositeLitDetails.AddField("Namespace", astbuilder.Selector(dst.NewIdent(receiverIdent), "Namespace"))
```
Comment (babbageclunk) [inline v2/tools/generator/internal/interfaces/kubernetes_resource_interface.go:403]: This can go now - presumably left from initial steps.

**Review by babbageclunk — APPROVED** [review]
Nice!

File: v2/tools/generator/internal/interfaces/kubernetes_resource_interface.go (hunk 2 above)
Comment (matthchr) [reply v2/tools/generator/internal/interfaces/kubernetes_resource_interface.go:403]: Oh, yep - good catch

File: v2/pkg/genruntime/resource_reference.go (hunk 1 above)
Comment (matthchr) [reply v2/pkg/genruntime/resource_reference.go:112]: Good prodding to add test, found some awkward behavior (not really a bug but confusing). Clarified it, added comment, and added test.

**Pull request merged**